app/controllers/integrations/ci_pipeline_controller.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+module Integrations
+  class CiPipelineController < Integrations::BaseController
+    private
+
+    def deploy?
+      true
+    end
+
+    def deploy
+      @deploy ||= params.fetch(:deploy)
```

Comment:
use `.require` to get nice error messages ?
(also below for commit/sha/message)

Reply:
How about something like a FormObject so we can use validations on that? Or else surround each method using `require` with a rescue so that we can give back 422s instead of blowing up with an exception?